Sources/PrometheusMetrics/PrometheusMetrics.swift

```diff
@@ -0,0 +1,307 @@
+            return createHandler(counter)
+        }
+        return createHandler(self.createCounter(forType: Int64.self, named: label, withLabelType: DimensionLabels.self))
+    }
+    
+    /// Makes a recorder
+    public func makeRecorder(label: String, dimensions: [(String, String)], aggregate: Bool) -> RecorderHandler {
+        return aggregate ? makeHistogram(label: label, dimensions: dimensions) : makeGauge(label: label, dimensions: dimensions)
+    }
+    
+    private func makeGauge(label: String, dimensions: [(String, String)]) -> RecorderHandler {
+        let createHandler = { (gauge: PromGauge) -> RecorderHandler in
+            return MetricsGauge(gauge: gauge, dimensions: dimensions)
+        }
+        if let gauge: PromGauge<Double, DimensionLabels> = self.getMetricInstance(with: label, andType: .gauge) {
+            return createHandler(gauge)
+        }
+        return createHandler(createGauge(forType: Double.self, named: label, withLabelType: DimensionLabels.self))
+    }
+    
+    private func makeHistogram(label: String, dimensions: [(String, String)]) -> RecorderHandler {
+        let createHandler = { (histogram: PromHistogram) -> RecorderHandler in
+            return MetricsHistogram(histogram: histogram, dimensions: dimensions)
+        }
+        if let histogram: PromHistogram<Double, DimensionHistogramLabels> = self.getMetricInstance(with: label, andType: .histogram) {
+            return createHandler(histogram)
+        }
+        return createHandler(createHistogram(forType: Double.self, named: label, labels: DimensionHistogramLabels.self))
+    }
+    
+    /// Makes a timer
+    public func makeTimer(label: String, dimensions: [(String, String)]) -> TimerHandler {
+        let createHandler = { (summary: PromSummary) -> TimerHandler in
+            return MetricsSummary(summary: summary, dimensions: dimensions)
+        }
+        if let summary: PromSummary<Int64, DimensionSummaryLabels> = self.getMetricInstance(with: label, andType: .summary) {
+            return createHandler(summary)
+        }
+        return createHandler(createSummary(forType: Int64.self, named: label, labels: DimensionSummaryLabels.self))
+    }
+}
+
+public extension MetricsSystem {
+    /// Get the bootstrapped `MetricsSystem` as `PrometheusClient`
+    ///
+    /// - Returns: `PrometheusClient` used to bootstrap `MetricsSystem`
+    /// - Throws: `PrometheusError.PrometheusFactoryNotBootstrapped`
+    ///             if no `PrometheusClient` was used to bootstrap `MetricsSystem`
+    static func prometheus() throws -> PrometheusClient {
+        guard let prom = self.factory as? PrometheusClient else {
+            throw PrometheusError.PrometheusFactoryNotBootstrapped
+        }
+        return prom
+    }
+}
+
+// MARK: - Labels
+
+/// A generic `String` based `CodingKey` implementation.
+private struct StringCodingKey: CodingKey {
+    /// `CodingKey` conformance.
```

Comment:
is it typical to comment on all conforming things?  I think I'd drop those comments hm...
Although I could be missing a Swifty pattern about this?

Reply:
I think it's for Jazzy so it doesn't show something low like "20% documented". But I haven't seen a comment that explains the conformance, so I'd replace that comment too.

If you want to link to the description of some other thing I think you can use `@see OtherThing.stringValue`

Reply:
It was indeed for Jazzy. But I can take another look at all comments and give them a bit more meaningful comments 😄